docs/error-handling.md

```diff
@@ -117,6 +117,11 @@ The public error API
   the last error.  This may return NULL if no error has occurred.
   Otherwise this should return a `git_error` object indicating the class
   of error and the error message that was generated by the library.
+  Do not use this function unless the prior call to a libgit2 API
+  returned an error, as it can give misleading results.
```

Comment:
"as it can otherwise give misleading results." Just to make sure that it may only be misleading in this exact case ;)